pkg/subctl/cmd/show_all.go

```diff
@@ -26,14 +26,23 @@ func showAll(cmd *cobra.Command, args []string) {
 		fmt.Println()
 		fmt.Printf("Showing information for cluster %q:\n", item.clusterName)
 
+		fmt.Println("Showing Network details")
 		showNetworkSingleCluster(item.config)
-		fmt.Println()
+		fmt.Println("")
+
+		submariner := GetSubmarinerResource(item.config)
+
+		if submariner == nil {
+			continue
+		}
+
+		fmt.Println("\nShowing Endpoint details")
 		showEndpointsFromConfig(item.config)
-		fmt.Println()
+		fmt.Println("\nShowing Connection details")
 		showConnectionsFromConfig(item.config)
-		fmt.Println()
+		fmt.Println("\nShowing Gateway details")
 		showGatewaysFromConfig(item.config)
```

Comment:
showGatewaysFor(submariner) will again check for submariner != nil (which will always be true) because L44 will be executed only if submariner != nil.

1 way to do it is `printGateways(getGatewaysStatus(submariner))` which I dont like much. Another way is to keep the current implementation as is.

Reply:
It's fine if `showGatewaysFor` checks it again as `submariner` may be nil from the other call site. The other alternative is to check nil at the other call site - either way, doesn't matter.